src/components/Bookmarks.vue

```diff
@@ -1,311 +1,443 @@
-                index: this.editableSectionIndex,
-                section: this.editableSection
-            });
-            this.closeEditModal();
-        },
+    newItem.value = { title: '', url: '', icon: '', iconType: 'font' }
+    closeModal()
+  } catch (err) {
+    error.value = err instanceof Error ? err.message : 'Failed to add item'
+  } finally {
+    isLoading.value = false
+  }
+}
 
-        updateItem() {
-            // Normalize the URL
-            this.editableItem.url = this.normalizeUrl(this.editableItem.url);
-            
-            this.$emit('update-item', {
-                sectionIndex: this.editableSectionIndex,
-                itemIndex: this.editableItemIndex,
-                item: this.editableItem
-            });
-            this.closeEditModal();
-        },
-        deleteSection() {
-            if (confirm('Are you sure you want to delete this section?')) {
-                this.$emit('delete-section', this.editableSectionIndex);
-                this.closeEditModal();
-            }
-        },
+const addSection = () => {
+  try {
+    clearError()
+    
+    const validation = validateSection(newSectionName.value)
+    if (validation) {
+      error.value = validation
+      return
+    }
+    
+    emit('add-section', newSectionName.value)
+    newSectionName.value = ''
+    showAddSectionModal.value = false
+  } catch (err) {
+    error.value = err instanceof Error ? err.message : 'Failed to add section'
+  }
+}
 
-        deleteItem() {
-            if (confirm('Are you sure you want to delete this item?')) {
-                this.$emit('delete-item', {
-                    sectionIndex: this.editableSectionIndex,
-                    itemIndex: this.editableItemIndex
-                });
-                this.closeEditModal();
-            }
-        },
-    },
-};
+const updateSection = () => {
+  try {
+    clearError()
+    
+    if (!editableSection.value) return
+    
+    const validation = validateSection(editableSection.value.name)
+    if (validation) {
+      error.value = validation
+      return
+    }
+    
+    emit('update-section', {
+      index: editableSectionIndex.value,
+      section: editableSection.value,
+    })
+    closeEditModal()
+  } catch (err) {
+    error.value = err instanceof Error ? err.message : 'Failed to update section'
+  }
+}
+
+const updateItem = async () => {
+  try {
+    clearError()
+    
+    if (!editableItem.value) return
+    
+    const validation = validateItem(editableItem.value)
+    if (validation) {
+      error.value = validation
+      return
+    }
+
+    isLoading.value = true
+    editableItem.value.url = normalizeUrl(editableItem.value.url)
+    
+    if (!editableItem.value.icon || editableItem.value.icon === 'fas fa-link') {
+      const mappedIcon = getIconForDomain(editableItem.value.url)
+      if (mappedIcon !== 'fas fa-link') {
+        editableItem.value.icon = mappedIcon
+        editableItem.value.iconType = 'font'
```

Comment:
iconType is always font execept when we got value from `fetchWebsiteIcon`. so     `editableItem.value.iconType = 'font'` can be the default